static/js/message_events.js

```diff
@@ -370,6 +370,12 @@ export function update_messages(events) {
                 } else if (!current_filter) {
                     message_lists.current.add_messages(event_messages);
                 } else {
+                    // Remove existing message that were updated since they may not be a part of the filter now. Also, this will
+                    // help up rerender them via maybe_add_narrowed_messages, if they were simply updated.
+                    const updated_messages = event_messages.filter(
+                        (msg) => message_lists.current.data.get(msg.id) !== undefined,
+                    );
+                    message_lists.current.data.remove(updated_messages.map((msg) => msg.id));
```

Comment:
Should this being using `message_lists.current.remove_and_rerender` instead?  Or is it intentional that this is a data-only change?

Reply:
(I would expect it to mostly not matter, because the thing we're about to do will often trigger a rerender, but if `maybe_add_narrowed_messages` returns nothing, I'd expect failing to rerender here to lead to a bug where things don't correctly get removed from the search view)

Reply:
> I'd expect failing to rerender here to lead to a bug where things don't correctly get removed from the search view

Fair point. Used `remove_and_rerender`